server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java

```diff
@@ -99,9 +104,16 @@ private void startWorkqueue() {
 
     @VisibleForTesting
     void declareQueue() {
-        channelPool.getSender().declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME)).block();
-        channelPool.getSender().declare(QueueSpecification.queue(QUEUE_NAME).durable(true).arguments(Constants.WITH_SINGLE_ACTIVE_CONSUMER)).block();
-        channelPool.getSender().bind(BindingSpecification.binding(EXCHANGE_NAME, ROUTING_KEY, QUEUE_NAME)).block();
+        channelPool.getSender()
+            .declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME))
+            .retryBackoff(NUM_RETRIES, FIRST_BACKOFF, FOREVER)
+            .then(channelPool.getSender()
+                .declare(QueueSpecification.queue(QUEUE_NAME).durable(true).arguments(Constants.WITH_SINGLE_ACTIVE_CONSUMER))
+                .retryBackoff(NUM_RETRIES, FIRST_BACKOFF, FOREVER))
+            .then(channelPool.getSender()
+                .bind(BindingSpecification.binding(EXCHANGE_NAME, ROUTING_KEY, QUEUE_NAME))
+                .retryBackoff(NUM_RETRIES, FIRST_BACKOFF, FOREVER))
+            .block();
```

Comment:
I prefer that you split management operation (create a variable) and `then`s-chaining/`block`.